event.go

```diff
@@ -219,6 +220,15 @@ func (e *Event) Bytes(key string, val []byte) *Event {
 	return e
 }
 
+// Hex adds the field key with val as a hex string to the *Event context.
+func (e *Event) Hex(key string, val []byte) *Event {
+	if e == nil {
+		return e
+	}
+	e.buf = json.AppendString(json.AppendKey(e.buf, key), hex.EncodeToString(val))
```

Comment:
You could create a `json.AppendHex` that just does:

```go
func AppendHex(buf, val []byte) []byte {
    const hextable = "0123456789abcdef"
    for _, v := range val {
        buf = append(buf, hextable[v>>4], hextable[v&0x0f])
    }
    return buf
}
```

This would avoid the slice heap allocation done by `hex.EncodeToString`.

Reply:
Done. I also added comments for all the `AppendXXX(...)` functions that were missing them to shut up the linter.